### ISSUE TITLE
New zarr requirement is not listed

### ISSUE THREAD

**Issue by QuLogic** [dask/dask#5406]
`zarr` is newly imported and used in tests (without skips), but it is not listed in `setup.py` anywhere. So I don't know if it was supposed to be a dependency for arrays, or for anything at all. Then tests without it [fail](https://koji.fedoraproject.org/koji/taskinfo?taskID=37655244).
```pytb
________________ test_compute_chunk_sizes_warning_fixes_to_zarr ________________
unknown = dask.array<getitem, shape=(nan,), dtype=float64, chunksize=(nan,), chunktype=numpy.ndarray>
    def test_compute_chunk_sizes_warning_fixes_to_zarr(unknown):
        y = unknown
        with pytest.raises(ValueError, match="compute_chunk_sizes"):
            with StringIO() as f:
>               y.to_zarr(f)
dask/array/tests/test_array_core.py:4183: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
dask/array/core.py:2296: in to_zarr
    return to_zarr(self, *args, **kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
arr = dask.array<getitem, shape=(nan,), dtype=float64, chunksize=(nan,), chunktype=numpy.ndarray>
url = <_io.StringIO object at 0x7f0fbd3e84c0>, component = None
storage_options = None, overwrite = False, compute = True, return_stored = False
kwargs = {}
    def to_zarr(
        arr,
        url,
        component=None,
        storage_options=None,
        overwrite=False,
        compute=True,
        return_stored=False,
        **kwargs
    ):
        """Save array to the zarr storage format
    
        See https://zarr.readthedocs.io for details about the format.
    
        Parameters
        ----------
        arr: dask.array
            Data to store
        url: Zarr Array or str or MutableMapping
            Location of the data. A URL can include a protocol specifier like s3://
            for remote data. Can also be any MutableMapping instance, which should
            be serializable if used in multiple processes.
        component: str or None
            If the location is a zarr group rather than an array, this is the
            subcomponent that should be created/over-written.
        storage_options: dict
            Any additional parameters for the storage backend (ignored for local
            paths)
        overwrite: bool
            If given array already exists, overwrite=False will cause an error,
            where overwrite=True will replace the existing data.
        compute, return_stored: see ``store()``
        kwargs: passed to the ``zarr.create()`` function, e.g., compression options
    
        Raises
        ------
        ValueError
            If ``arr`` has unknown chunk sizes, which is not supported by Zarr.
    
        See Also
        --------
        dask.array.Array.compute_chunk_sizes
    
        """
>       import zarr
E       ModuleNotFoundError: No module named 'zarr'
dask/array/core.py:2859: ModuleNotFoundError
```

**Comment by jrbourbeau** [comment]
Thanks for catching this @QuLogic! We should make sure this test is skipped if `zarr` isn't installed by adding `pytest.importorskip("zarr")` to the top of the test. Any interest in submitting a PR with a fix?

**Comment by QuLogic** [comment]
Sure, that's easy; should this be listed in `setup.py` somewhere though?

**Comment by jrbourbeau** [comment]
I think we want to avoid adding `zarr` to `setup.py`. Since `zarr` is only necessary for `to_zarr`/`from_zarr` and isn't required more broadly for using `dask.array` like, e.g. `toolz` is, we don't want to make all `dask.array` users install `zarr` unnecessarily and instead treat it as an optional dependency for those who would like to use `zarr` I/O features. If you're curious, there was a recent conversation on this topic over in #5209.

**Comment by jcrist** [comment]
I forgot I added a fix for this in #5400 while cleaning up the PY2 code. Tests now should pass without `zarr` installed. Closing.
